ansible/playbooks/jenkins/worker/create.yml

```diff
@@ -97,20 +94,25 @@
       - not os|startswith("win")
       - not os|startswith("zos")
       - os != "macos11.0"
+      - not (alias is defined and alias is match('jenkins-workspace.*'))
       package:
         name: "{{ package }}"
         state: absent
       loop_control:
         loop_var: package
       with_items: [ "node", "nodejs", "npm" ]
     - name: fail if node is in the path - please uninstall it
-      when: not os|startswith("win")
+      when:
+      - not os|startswith("win")
+      - not (alias is defined and alias is match('jenkins-workspace.*'))
       shell: "node -v"
       register: node_check_cmd
       failed_when: node_check_cmd.rc == 0
       changed_when: False
     - name: fail if node is in the path - please uninstall it
-      when: os|startswith("win")
+      when:
+      - os|startswith("win")
+      - not (alias is defined and alias is match('jenkins-workspace.*'))
```

Comment:
Technically this one isn't needed -- we have no "win*" hosts that are also `jenkins-workspace` but I've added it for consistency with all the other tasks.